tests/lib/prepare.sh

```diff
@@ -475,6 +475,23 @@ uc20_build_initramfs_kernel_snap() {
         echo "if test -d /run/mnt/data/system-data; then touch /run/mnt/data/system-data/the-tool-ran; fi" >> \
             "$skeletondir/main/usr/lib/the-tool"
 
+
+        # patch the initramfs to go back to isolating the initrd units and to 
+        # not specify to mount /run/mnt/ubuntu-boot via fstab, these were all 
+        # done as interim changes until snap-bootstrap took control of things,
+        # now that snap-bootstrap is in control, we don't want those hacks, but 
+        # we still want accurate test results so drop those hacks to let 
+        # snap-bootstrap do everything for the spread run
+
+        # TODO:UC20: drop these patches when the associated changes have landed
+        #            upstream
+        rm -rf "$skeletondir/main/usr/lib/systemd/system/initrd-cleanup.service.d/core-override.conf"
+        sed -i "$skeletondir/main/usr/lib/systemd/system/populate-writable.service" \
+            -e "s@ExecStartPost=/usr/bin/systemctl --no-block start initrd.target@ExecStartPost=/usr/bin/systemctl --no-block isolate initrd.target@"
+        sed -i "$skeletondir/main/usr/lib/the-modeenv" \
+            -e "s@echo 'LABEL=ubuntu-boot /run/mnt/ubuntu-boot auto defaults 0 0' >> /run/image.fstab@echo not doing anything@"
```

Comment:
+1 on dropping boot mounts from scripts.

But not sure about return to isolation. As writable-path mounts are not hooked up to the initrd.targets, or are they?

Reply:
Hmm, I will need to look into writable-path mounts a bit to see what implications there are for that. Ideally, writable-paths would handle that independently from snap-bootstrap I think

Reply:
I was looking at this today - for most mounts this should be fine as they will not be done when initramfs runs, cf.:
https://github.com/snapcore/core20/blob/master/static/usr/lib/core/handle-writable-paths#L124

But I think we need to investigate the case where the mounts are in "/etc/" - this includes /etc/systemd itself which is the reason why the mounting is done early. 

This actually makes me wonder if as a followup we should simply implement parsing/generating the writable-path from snap-bootstrap WDYT?